src/App.jsx

```diff
@@ -5,19 +5,32 @@ import Home from './components/Home';
 import Products from './components/Products';
 import ProductDetail from './components/ProductDetail';
 import Toolbar from './components/Toolbar';
-import axios from "axios";
+import { ThemeContext, themes } from './ThemeContext';
+import ToggleSwitch from './components/ToggleSwitch';
+import { useContext, useEffect, useState } from 'react';
+
+
 
 function App() {
+
+  const{toggleTheme, theme} = useContext(ThemeContext); 
+  
+  useEffect(()=>{
+    document.body.style.background = theme.background
+    }, [theme]);
+  
   return (
     <Router>
       <Toolbar />
-      <Switch>
-        <Route exact path="/" component={Home} />
-        <Route exact path="/products" component={Products} />
-        <Route path="/products/:id" component={ProductDetail} />
-      </Switch>
+      <ToggleSwitch onToggle={toggleTheme} />
+        <Switch>
```

Comment:
`Switch` is for an older version of React Router. You should be using `<Routes>` like did in class or have on Canvas. It's important to understand which version you are using and when you look up content elsewhere, and that content is relevant to the version. I always recommend looking at the official docs.

You can find the version my looking at `package.json`